src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -2222,7 +2249,7 @@ skip:
       NOTIFY_REQUEST_GET_OBJECTS::request req;
       bool is_next = false;
       size_t count = 0;
-      const size_t count_limit = m_core.get_block_sync_size(m_core.get_current_blockchain_height());
+      m_bss = m_core.get_block_sync_size(m_core.get_current_blockchain_height());
```

Comment:
I would also make this is an atomic, I think it works fine if so (and then you'll have to cache the atomic read in the other function too otherwise it could change).

Reply:
Good suggestions. Thanks. Will fix this in updated PR.